app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -59,28 +53,30 @@ public void onEntryNameSubmitted() {
     }
 
     @Override
-    public void setContestName(String contestName) {
-        contestantWelcomeTextView.setText(getResources().getString(R.string.contestant_welcome_message, contestName));
+    public void showWelcomeMessage() {
+        setActionBarTitle(getResources().getString(R.string.contestant_welcome_message));
     }
 
     @Override
-    public void showEntryNameButton() {
+    public void enableEntryNameButton() {
+        entryNameNextButton.setEnabled(true);
         entryNameNextButton.setVisibility(android.view.View.VISIBLE);
     }
 
     @Override
-    public void hideEntryNameButton() {
+    public void disableEntryNameButton() {
+        entryNameNextButton.setEnabled(false);
         entryNameNextButton.setVisibility(android.view.View.GONE);
     }
 
     @Override
-    public void enableEntryNameButton() {
-        entryNameNextButton.setEnabled(true);
+    public void showIcon() {
+        entryNameIcon.setVisibility(android.view.View.VISIBLE);
```

Comment:
The problem is that I'm importing `import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;` statically, and if I import `android.view`, they conflict...
in the beginning Abass and I defined we'd always import the Contract statically, but now it might be time to rediscuss!
@abassawo